### PR TITLE
add googleSubjectId in ldap schema as an optional field

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadinstitute.atlassian.net/browse/GAWB-3760

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by gpolumbo-broad — APPROVED** [review]

**Review by MatthewBemis — COMMENTED** [review]
i know this is a very small piece of the overall change but is there anything that needs to be done to the pet schema in order to ensure that those will still be able to get thru the proxy?

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/schema/JndiSchemaDAO.scala [hunk 1]
```diff
@@ -256,6 +258,8 @@ class JndiSchemaDAO(protected val directoryConfig: DirectoryConfig, val schemaLo
 
     val may = new BasicAttribute("MAY")
     may.add(Attr.proxyEmail)
+    may.add(Attr.googleSubjectId)
+
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/schema/JndiSchemaDAO.scala:262]: Why is it optional?

Comment (Qi77Qi) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/schema/JndiSchemaDAO.scala:262]: @dvoet for invite endpoint, we don't know the invited user's googlesubjectId yet when it's created

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/schema/JndiSchemaDAO.scala [hunk 2]
```diff
@@ -418,6 +423,10 @@ class JndiSchemaDAO(protected val directoryConfig: DirectoryConfig, val schemaLo
     must.add(Attr.uid)
     attrs.put(must)
 
+    val may = new BasicAttribute("MAY")
+    may.add(Attr.googleSubjectId) //TODO: move this to must once old data is back populated
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/schema/JndiSchemaDAO.scala]: Ldap isn’t strict like a rdbms. I think you can make it must now.

**Review by dvoet — APPROVED** [review]

**Comment by Qi77Qi** [comment]
jenkins retest

**Pull request merged**